tests/integration_tests/style/test_gitlint.py

```diff
@@ -14,6 +15,23 @@ def test_gitlint():
     """
     os.environ['LC_ALL'] = 'C.UTF-8'
     os.environ['LANG'] = 'C.UTF-8'
+    # devtool script will set the working directory to FC_ROOT/tests.
+    # We will need the parent for safe.directory configuration
+    working_dir = Path(os.getcwd())
+
+    # The firecracker root git folder in the container is
+    # bind-mounted to a folder on the host which is mapped to a
+    # user that is different from the user which runs the integ tests.
+    # This difference in ownership is validated against by git.
+    # https://github.blog/2022-04-12-git-security-vulnerability-announced/
+    try:
+        utils.run_cmd('git config --global '
+                      '--add safe.directory {}'.format(working_dir.parent))
```

Comment:
I was able to run the gitlint test manually to check that this worked as expected. However there are some other tests that are failing that may be related to the broader Ubuntu upgrade. It would be my preference that config like this only applies to tests that require it. I will need to look at those tests a bit more first to make sure that the approach of upgrading Ubuntu to achieve the Python upgrade is viable. If I only upgrade Python specifically then this git/gitlint issue may not be relevant anymore.